watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,460 @@
+        hydraulic_diameter_method=HydraulicDiameterMethod.spacer_specific_area_known,
+        friction_factor_method=FrictionFactorMethod.Gurreri,
+    )
+    m.fs.sepa1 = Separator(
+        property_package=m.fs.properties,
+        outlet_list=["to_disp", "to_conc_in1"],
+    )
+    m.fs.prod = Product(property_package=m.fs.properties)
+    m.fs.disp = Product(property_package=m.fs.properties)
+
+    # touch necessary variables to ensure they are constructed and solved
+    m.fs.prod.properties[0].flow_vol_phase[...]
+    m.fs.disp.properties[0].flow_vol_phase[...]
+
+    # costing
+    m.fs.EDstack.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.pump0.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.pump1.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.costing.cost_process()
+    m.fs.costing.add_annual_water_production(
+        m.fs.prod.properties[0].flow_vol_phase["Liq"]
+    )
+    m.fs.costing.add_LCOW(m.fs.prod.properties[0].flow_vol)
+    m.fs.costing.add_specific_energy_consumption(
+        m.fs.prod.properties[0].flow_vol_phase["Liq"]
+    )
+
+    # add extra variables and constraints
+    m.fs.recovery_vol_H2O = Var(
+        initialize=0.5,
+        bounds=(0, 1),
+        domain=NonNegativeReals,
+        units=pyunits.dimensionless,
+        doc="flowsheet level water recovery calculated by volumeric flow rate",
+    )
+    m.fs.mem_area = Var(
+        initialize=1,
+        bounds=(0, 1e3),
+        units=pyunits.meter**2,
+        doc="Total membrane area for cem (or aem) in one stack",
+    )
+    m.fs.prod_salinity = Var(
+        initialize=1, bounds=(0, 1000), units=pyunits.kg * pyunits.meter**-3
+    )
+    m.fs.disp_salinity = Var(
+        initialize=1, bounds=(0, 1e6), units=pyunits.kg * pyunits.meter**-3
+    )
+    m.fs.eq_recovery_vol_H2O = Constraint(
+        expr=m.fs.recovery_vol_H2O
+        == m.fs.prod.properties[0].flow_vol_phase["Liq"]
+        * m.fs.feed.properties[0].flow_vol_phase["Liq"] ** -1
+    )
+    m.fs.eq_electrodialysis_equal_flow = Constraint(
+        expr=m.fs.EDstack.diluate.properties[0, 0].flow_vol_phase["Liq"]
+        == m.fs.EDstack.concentrate.properties[0, 0].flow_vol_phase["Liq"]
+    )
+
+    m.fs.eq_product_salinity = Constraint(
+        expr=m.fs.prod_salinity
+        == sum(
+            m.fs.prod.properties[0].conc_mass_phase_comp["Liq", j]
+            for j in m.fs.properties.ion_set | m.fs.properties.solute_set
+        )
+    )
+    m.fs.eq_disposal_salinity = Constraint(
+        expr=m.fs.disp_salinity
+        == sum(
+            m.fs.disp.properties[0].conc_mass_phase_comp["Liq", j]
+            for j in m.fs.properties.ion_set | m.fs.properties.solute_set
+        )
+    )
+
+    m.fs.eq_mem_area = Constraint(
+        expr=m.fs.mem_area
+        == m.fs.EDstack.cell_width
+        * m.fs.EDstack.cell_length
+        * m.fs.EDstack.cell_pair_num
+    )
```

Comment:
Given this constraint, this reinforces my previous question on why this is at the flowsheet level instead of the unit level.

Reply:
I still think you should have `area` on the unit model; that would make it consistent with other unit models that we have. Even if you can get away without it, I'd rather try to keep some level of consistency where possible.

Reply:
I consider an optimal model structure more important than "some level of consistency".  The `area` isn't an essential variable member for ED model but maybe it is for other models.  Adding it means adding a reducible dimension to the equation matrix.  The solver will take resource to reduce the dimension.  When building complex flowsheet connecting multiple unit models, maintaining each unit in its cleanest model structure will be in favor for minimizing numerical difficulties in all aspects.   Secondary things like this are more flexibly handled at the flowsheet level than being embedded in unit models.   Please see if this change your opinion.  If not, I will need more time to make this change in a couple of files.

Reply:
I think we should either consider (1) making area an `Expression` on the unit model or (2) making it an `Expression` on the flowsheet.

I suspect that goes for all of these quantities. No reason for a variable just for reporting purposes.